src/NetAppFiles/NetAppFiles/Volume/NewNetAppFilesVolume.cs

```diff
@@ -0,0 +1,149 @@
+
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = FieldsParameterSet,
+            HelpMessage = "The location of the resource")]
+        [ValidateNotNullOrEmpty]
+        [LocationCompleter("Microsoft.NetApp/netAppAccounts/capacityPools/volumes")]
+        public string Location { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = FieldsParameterSet,
+            HelpMessage = "The name of the ANF account")]
+        [ValidateNotNullOrEmpty]
+        [ResourceNameCompleter(
+            "Microsoft.NetApp/netAppAccount",
+            nameof(ResourceGroupName))]
+        public string AccountName { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = FieldsParameterSet,
+            HelpMessage = "The name of the ANF pool")]
+        [ValidateNotNullOrEmpty]
+        [ResourceNameCompleter(
+            "Microsoft.NetApp/netAppAccounts/capacityPools",
+            nameof(ResourceGroupName),
+            nameof(AccountName))]
+        public string PoolName { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The name of the ANF volume")]
+        [ValidateNotNullOrEmpty]
+        [Alias("VolumeName")]
+        [ResourceNameCompleter(
+            "Microsoft.NetApp/netAppAccounts/capacityPools/volumes",
+            nameof(ResourceGroupName),
+            nameof(AccountName),
+            nameof(PoolName))]
+        public string Name { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The maximum storage quota allowed for a file system in bytes")]
+        [ValidateNotNullOrEmpty]
+        public long? UsageThreshold { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The Azure Resource URI for a delegated subnet")]
+        [ValidateNotNullOrEmpty]
+        public string SubnetId { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "A unique file path for the volume")]
+        [ValidateNotNullOrEmpty]
+        public string CreationToken { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The service level of the ANF volume")]
+        [ValidateNotNullOrEmpty]
+        [PSArgumentCompleter("Standard", "Premium", "Extreme")]
+        public string ServiceLevel { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "A hashtable which represents resource tags")]
+        [ValidateNotNullOrEmpty]
+        [Alias("Tags")]
+        public Hashtable Tag { get; set; }
+
+        [Parameter(
+            ParameterSetName = ParentObjectParameterSet,
+            Mandatory = false,
+            ValueFromPipeline = true,
+            HelpMessage = "The pool for the new volume object")]
+        [ValidateNotNullOrEmpty]
+        public PSNetAppFilesPool InputObject { get; set; }
```

Comment:
@leonardbf nit: please rename this to `PoolObject`

Reply:
Ok. Renamed.